### PR TITLE
fix(lsp): include border in float height

### LINKED ISSUE #25521: bug: `open_floating_preview` position doesn't take into account border
### Problem

w/o border:
Screenshot 2023-10-06 at 3 54 57 AM

w/ border:
Screenshot 2023-10-06 at 3 57 57 AM

Seems like we just need to add an offset for border width. There's some similar logic in nvim-cmp that handles this correctly.

### Steps to reproduce

No minimal config yet, but it involves

```lua
vim.lsp.handlers["textDocument/hover"] =
  vim.lsp.with(vim.lsp.handlers.hover, {
    border = "single",
  })
```

I was thinking of looking at this myself, but if I don't get around to that then I'll post a full minimal config

### Expected behavior

Floating window should not cover cursor

### Neovim version (nvim -v)

f67517bba

### Vim (not Nvim) behaves the same?

no floating windows i think

### Operating system/version

macOS 13.4.1

### Terminal name/version

kitty 0.28.1

### $TERM environment variable

xterm-kitty

### Installation

build from repo

### PR DESCRIPTION
**Pull request by MariaSolOs**
Fixes #25521

### CODE REVIEW

**Comment by glepnir** [comment]
I prefer #25539

File: runtime/lua/vim/lsp/util.lua [hunk 1]
```diff
@@ -26,7 +26,7 @@ local default_border = {
 --- size it adds to a float.
 ---@param opts table optional options for the floating window
 ---            - border (string or table) the border
----@return table size of border in the form of { height = height, width = width }
+---@return { height: integer, width: integer }
```
Comment (llllvvuu) [inline runtime/lua/vim/lsp/util.lua:29]: This is a good catch though, should still get this in (maybe in https://github.com/neovim/neovim/pull/25171)

**Review by llllvvuu — COMMENTED** [review]
Sorry about the awkward timing on this haha.

FWIW this wouldn't have actually fixed the issue for a not-immediately-obvious reason: this function isn't actually responsible for truncating the height. It's pretty funky stuff that could probably be refactored somehow but my brain is a bit too 🥴 right now

**Comment by MariaSolOs** [comment]
> Sorry about the awkward timing on this haha.
> 
> FWIW this wouldn't have actually fixed the issue for a not-immediately-obvious reason: this function isn't actually responsible for truncating the height. It's pretty funky stuff that could probably be refactored somehow but my brain is a bit too 🥴 right now

Oh no worries! And yes I realized that this wasn’t a complete fix and hence why I converted it to a draft.

**Pull request closed without merging**